src/screens/IndexScreen.js

```diff
@@ -367,6 +369,40 @@ export const IndexScreen = ({ navigation, route }) => {
 
   const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
 
+  useLayoutEffect(() => {
+    if (query === QUERY_TYPES.POINTS_OF_INTEREST && showMap) {
+      navigation.setOptions({
+        headerLeft: () => (
+          <HeaderLeft
+            onPress={() => {
+              const selectedFilter = filterType.find((entry) => entry.selected);
+              const updatedFilter = filterType.map((entry) => {
+                if (entry.id !== selectedFilter.id) {
+                  entry.selected = true;
+                } else {
+                  entry.selected = false;
+                }
+                return entry;
+              });
```

Comment:
this will not work if the filters are more than two, because then more than 1 would be set to true here, as only 1 can be selected at the same time.
so thinking again, what do we want here. we want to "jump" in the previous screen view which is the same as before. we want to kind of "rest" the header left situation. in this case, we do not need to have any logics but to set the initial state: `onPress={() => setFilterType(INITIAL_FILTER)}`